crates/plugins/src/store.rs

```diff
@@ -0,0 +1,117 @@
+use anyhow::{anyhow, Context, Result};
+use flate2::read::GzDecoder;
+use std::{
+    fs::{self, File},
+    path::{Path, PathBuf},
+};
+use tar::Archive;
+
+use crate::manifest::PluginManifest;
+
+/// Directory where the manifests of installed plugins are stored.
+pub const PLUGIN_MANIFESTS_DIRECTORY_NAME: &str = "manifests";
+
+/// Houses utilities for getting the path to Spin plugin directories.
+pub struct PluginStore {
+    root: PathBuf,
+}
+
+impl PluginStore {
+    pub fn new(root: impl AsRef<Path>) -> Self {
+        Self {
+            root: root.as_ref().to_owned(),
```

Comment:
As a general pattern:

```suggestion
    pub fn new(root: impl Into<PathBuf>) -> Self {
        Self {
            root: root.into(),
```